paasta_tools/cli/cmds/status.py

```diff
@@ -327,6 +337,60 @@ def status_kubernetes_job_human(
         )
 
 
+def print_flinkcluster_status(
+    service: str,
+    instance: str,
+    output: List[str],
+    status,
+    verbose: int,
+) -> int:
+    if verbose:
+        output.append(f"    Flink version: {status.config['flink-version']} {status.config['flink-revision']}")
+    else:
+        output.append(f"    Flink version: {status.config['flink-version']}")
+    output.append(
+        "    Jobs:"
+        f" {status.overview['jobs-running']} running,"
+        f" {status.overview['jobs-finished']} finished,"
+        f" {status.overview['jobs-failed']} failed,"
+        f" {status.overview['jobs-cancelled']} cancelled",
+    )
+    output.append(
+        "   "
+        f" {status.overview['taskmanagers']} taskmanagers,"
+        f" {status.overview['slots-available']}/{status.overview['slots-total']} slots available",
+    )
+
+    output.append(f"    Jobs:")
+    if verbose:
+        output.append(f"      Job Name                         State       Job ID                           Started")
+    else:
+        output.append(f"      Job Name                         State       Started")
+    for job in status.jobs:
+        job_id = job['jid']
+        if verbose:
+            fmt = "      {job_name: <32.32} {state: <11} {job_id} {start_time}"
+        else:
+            fmt = "      {job_name: <32.32} {state: <11} {start_time}"
+        start_time = datetime_from_utc_to_local(datetime.utcfromtimestamp(int(job['start-time']) // 1000))
+        output.append(fmt.format(
+            job_id=job_id,
+            job_name=job['name'].split('.', 2)[2],
+            state=job['state'],
+            start_time=f'{str(start_time)} ({humanize.naturaltime(start_time)})',
+        ))
+        if job_id in status.exceptions:
+            exceptions = status.exceptions[job_id]
+            root_exception = exceptions['root-exception']
+            if root_exception is not None:
```

Comment:
can there be not-None `exceptions` without root-exception?

Reply:
Nope
There is always `"root-exception"` key in `exceptions`, please see the flink operator src for details.

Reply:
then we  should probably change this `if` to something like

    if root_eception  is None:
      log.error()
      return

Reply:
There are two cases for the `exceptions` field:
1. a job without exception:
```yaml
  exceptions:
    2f9a6671abd1894d794262ef14a1bf3e:
      root-exception: null
      timestamp: null
```
2. a job with an exception:
```yaml
  exceptions:
    2f9a6671abd1894d794262ef14a1bf3e:
      root-exception: "somethingsomething"
      timestamp: 1551164324119
```
Why we should call `log.error()` and stop processing if `root_exception` is `None` (or even if it isn't `None`)?

Reply:
i  guess i misunderstood you when you said it's impossible to have `exceptions`  field without `root-exception`. for me having `root-exception: nil` means not having `root-exception` :)

is there a reason we include an entry with empty `root-exception` at all?

Reply:
This is Fink , not us.
See https://ci.apache.org/projects/flink/flink-docs-stable/monitoring/rest_api.html

Reply:
maybe doing omit-empty on exception  related  fields would make  golang serializer skip empty  maps? anyway, not related to this PR.

Reply:
At the moment values in `Status` have the same types as corresponding Flink REST API responses.
IMO we better continue using those types.